infrastructure/src/main/java/org/corfudb/infrastructure/RemoteMonitoringService.java

```diff
@@ -666,30 +672,43 @@ private void correctWrongEpochs(PollReport pollReport) {
             }
 
             // Update local layout copy.
-            serverContext.saveManagementLayout(latestLayout.get());
-            layout = serverContext.copyManagementLayout();
-
-            // In case of a partial seal, a set of servers can be sealed with a higher epoch.
-            // We should be able to detect this and bring the rest of the servers to this epoch.
-            long maxOutOfPhaseEpoch = Collections.max(wrongEpochs.values());
-            if (maxOutOfPhaseEpoch > layout.getEpoch()) {
-                layout.setEpoch(maxOutOfPhaseEpoch);
-            }
+            Layout newManagementLayout = latestLayout.get();
+            serverContext.saveManagementLayout(newManagementLayout);
 
-            // Re-seal all servers with the latestLayout epoch.
-            // This has no effect on up-to-date servers. Only the trailing servers are caught up.
-            getCorfuRuntime()
-                    .getLayoutView()
-                    .getRuntimeLayout(layout)
-                    .sealMinServerSet();
+            sealMinServerSet(wrongEpochs, newManagementLayout);
 
             // Check if any layout server has a stale layout.
             // If yes patch it (commit) with the latestLayout.
-            updateTrailingLayoutServers(layoutCompletableFutureMap);
+            updateTrailingLayoutServers(newManagementLayout, layoutCompletableFutureMap);
+            return newManagementLayout;
 
         } catch (QuorumUnreachableException e) {
             log.error("Error in correcting server epochs", e);
         }
+
+        return serverContext.copyManagementLayout();
+    }
+
+    private void sealMinServerSet(Map<String, Long> wrongEpochs, Layout managementLayout) {
```

Comment:
Can we come up with a better name?

Reply:
I am ok with this name. Please, any Ideas, happy to rename

Reply:
How about `sealWithLatestLayout` or `resealWithLatestKnownLayout`?